### PR TITLE
PSFR-1231 Add yaml config for assessments PoC

### PR DESCRIPTION
**Pull request by JamesBoobier**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsresettlementpassportapi/data/resettlementassessment/Question.kt [hunk 1]
```diff
@@ -1,21 +1,5 @@
...
-@JsonTypeInfo(
-  use = JsonTypeInfo.Id.NAME,
-  include = JsonTypeInfo.As.PROPERTY,
-  property = "@class",
-)
-@JsonSubTypes(
-  JsonSubTypes.Type(value = AccommodationResettlementAssessmentQuestion::class),
-  JsonSubTypes.Type(value = FinanceAndIdResettlementAssessmentQuestion::class),
-  JsonSubTypes.Type(value = GenericResettlementAssessmentQuestion::class),
-)
```
Comment (JamesBoobier) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsresettlementpassportapi/data/resettlementassessment/Question.kt:18]: This field isn't used by the UI so should be safe to remove from any json returned from the API.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsresettlementpassportapi/service/resettlementassessmentstrategies/AssessmentUtils.kt [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    }
+  } else {
+    throw ServerWebInputException("Support need [$supportNeed] must be a StringAnswer")
+  }
+}
+
+internal fun convertFromStringAnswer(answer: Answer<*>?): String {
+  if (answer is StringAnswer) {
+    return answer.answer ?: throw ServerWebInputException("Answer [$answer] must not be null")
+  } else {
```
Comment (riccardo-noviello-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsresettlementpassportapi/service/resettlementassessmentstrategies/AssessmentUtils.kt:27]: you don't need the else if the previous statement is returning

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsresettlementpassportapi/service/resettlementassessmentstrategies/AbstractResettlementAssessmentStrategy.kt [hunk 3]
```diff
@@ -40,8 +40,22 @@ abstract class AbstractResettlementAssessmentStrategy<T, Q>(
...
+    return if (useYaml) {
+      if (pathway == Pathway.ACCOMMODATION) {
+        false
+      } else {
+        pathway == this.pathway
+      }
+    } else {
+      pathway == this.pathway
+    }
+  }
```
Comment (riccardo-noviello-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsresettlementpassportapi/service/resettlementassessmentstrategies/AbstractResettlementAssessmentStrategy.kt:58]: too many else statements I think this is simpler
```
      if (useYaml && pathway == Pathway.ACCOMMODATION) {
        return false
      } 
      return pathway == this.pathway
    ```

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsresettlementpassportapi/integration/LegacyResettlementAssessmentIntegrationTest.kt [hunk 4]
```diff
@@ -0,0 +1,319 @@
...
+      .expectBody().json(
+        """
+        {
+          "nextPageId": "ISSUES_WITH_GAMBLING"
+        }
+        """.trimIndent(),
+      )
+  }
+
+  @Test
```
Comment (riccardo-noviello-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsresettlementpassportapi/integration/LegacyResettlementAssessmentIntegrationTest.kt:64]: I understand most of these tests are integration tests, quite difficult to understand (for someone like me) although the functionality implemented here could be unit tested quite easily, I don't know if it is an advantage or disadvantage to have this imbalance of integration tests vs unit going forward...just a consideration

**Review by riccardo-noviello-moj — APPROVED** [review]
Approving with personal considerations, nothing wrong with the PR really

**Pull request merged**